packages/common/src/models/index.ts

```diff
@@ -159,7 +159,9 @@ export function updateModel(
   return updateItem(opts as IUpdateItemOptions).then(() => {
     // To ensure we have the exact modified timestamp, we need to
     // get the item again
-    return getModel(item.id, requestOptions);
+    return item.data
+      ? getModel(item.id, requestOptions)
+      : getItem(item.id, requestOptions).then((i) => ({ item: i }));
```

Comment:
Ok, my  comment here wasn't saved, but I am worried about the fact that we are introducing this change. `updateModel`  is used in a variety of places so we need to have a pretty solid reason to change its behavior.

Reply:
I understand your concern, but I suggested this change. Up until now every item type we've thrown at `updateModel` has not only had `data`, but also known that the data was JSON. With this PR, we are now handling item types like PDF, that have blob data and if you call `getModel` it _always_ tries to fetch the item data and parse it as JSON.

To me the only risk is using the existence if `item.data` as a signal that we should should fetch the data. My assumption is that it will _always_ be truthy for things that have JSON data b/c they will have passed it in even if they did not make any changes. If that assumption is wrong, then we may need to think of another way to tell this fn when to use `getModel`, like adding a `fetchModel = true` to the options that are passed in. That would be safer, but more cumbersome, and I'm not sure it's needed.

Reply:
Another way to do deal w/ this is to have `getModel()` support types that don't have JSON data. I don't want to get into the business of basing that on `item.type`, but we might be able to do something like having it catch JSON parse errors, but then we'd be suppressing actual error's in the item's data.

@abp6318 - either way, at the very least we need to add a comment here saying that we cannot just call `getModel` as it stands today, b/c the content entity needs to support types that may not have JSON data.